### PR TITLE
Fix for DP-5 (Concurrent fetch() method calls trigger getData() multiple times)

### PR DESCRIPTION
**Pull request by radoh**

### CODE REVIEW

File: test/withDataProviders.test.js [hunk 1]
```diff
@@ -69,6 +69,23 @@ test('nested withDataProviders with eager prefetching - child component is rende
   expect(renderedMessage.textContent).toBe('1')
 })
 
+test('nested withDataProviders with eager prefetching - child component is rendered immediately, but ' +
+  'getData is called only once', async () => {
+  const getData = [getDataWithCount, {data: ''}, GET_DATA_DELAY]
+  const MessageContainer = messageContainer({needed: true, getData})
+  const ParentContainer = () => (<MessageContainer />)
```
Comment (jurom) [inline test/withDataProviders.test.js]: Why do you need this `ParentContainer`? Can't the `Container` decorate `MessageContainer` directly?

File: test/withDataProviders.test.js [hunk 2]
```diff
@@ -69,6 +69,23 @@ test('nested withDataProviders with eager prefetching - child component is rende
...
+  const Container = compose(
+    withDataProviders(() => [messageProvider({needed: false, getData})]),
+  )(ParentContainer)
+  const {root} = renderApp(<Container />)
+
+  await safeDelay(GET_DATA_DELAY)
```
Comment (jurom) [inline test/withDataProviders.test.js]: I can't comment the line, but now that I see this I realized that there probably shouldn't be `safeDelay` on line 65 in previous test - it should test that after the data is eagerly pre-fetched, the needed child is displayed immediately.

File: test/withDataProviders.test.js (hunk 1 above)
Comment (radoh) [reply test/withDataProviders.test.js]: Ah, you're right, I was copy-pasting this from a previous test, where there was a `show` parameter, which I didn't need here, and I didn't realize I no longer need `ParentContainer` at all...

File: test/withDataProviders.test.js (hunk 2 above)
Comment (radoh) [reply test/withDataProviders.test.js]: That's right, good catch, updated

**Pull request merged**